### DISCUSSION TITLE
Clickhouse error Cannot parse 'flow-ZUYGDTE3EBIXX352XPM3YEEFV4.proto'

### DISCUSSION THREAD

**Discussion by charlyforot** [akvorado/akvorado#447] [Q&A] [answered]
Hello,

I installed a fresh installation (compilation from source).

```
akvorado : 1.7.1
clickhouse : 22.8
kafka : 3.3.1
Go : 1.19.5
NodeJS : 16.19.0
npm : 8.19.3
```

A problem appears in Clickhouse about protobuf schema :

```
name:               CANNOT_PARSE_PROTOBUF_SCHEMA
code:               434
value:              173
last_error_time:    2023-01-31 10:15:16
last_error_message: Cannot parse 'flow-ZUYGDTE3EBIXX352XPM3YEEFV4.proto' file, found an error at line -1, column 0, File not found.
last_error_trace:   [171900250,367229284,459666884,459790936,459790339,367225945,367225649,367237058,367241074,365453758,338547398,338560911,365617205,365616006,365733386,365685627,365681369,365680861,365675692,338506333,338498303,334399064,334409526,334412908,172693672,172706493,140605812108967,140605811190319]
remote:             0
```

So there is no data in Clickhouse.

The connexion Kafka-Clickhouse looks like OK.

```
GROUP           TOPIC                            PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                                                                                HOST            CLIENT-ID
clickhouse      flows-ZUYGDTE3EBIXX352XPM3YEEFV4 0          -               486768          -               ClickHouse-akvorado-test-default-flows_ZUYGDTE3EBIXX352XPM3YEEFV4_raw-8ca4ca4e-42f0-406a-9f19-f3725ddb0885 /127.0.0.1      ClickHouse-akvorado-test-default-flows_ZUYGDTE3EBIXX352XPM3YEEFV4_raw
```

However, data seems to move from akvorado inlet to Kafka :

```
Topic flows-ZUYGDTE3EBIXX352XPM3YEEFV4 [0] at offset 509374: key ��Ӻ: 1675160391439
Topic flows-ZUYGDTE3EBIXX352XPM3YEEFV4 [0] at offset 509375: key %��: 1675160391439
Topic flows-ZUYGDTE3EBIXX352XPM3YEEFV4 [0] at offset 509376: key $��E: 1675160391439
Topic flows-ZUYGDTE3EBIXX352XPM3YEEFV4 [0] at offset 509377: key H�1: 1675160391439
```

I assure that sFlow packets are correctly formed.

Does anyone have an idea of the source of the problem ?

Thank you for your help !

**Comment by vincentbernat** [comment]
Try to restart the clickhouse container. It seems it didn't got the proto file from the orchestrator.

**Reply by charlyforot** [reply]
I have already restarted the clickhouse service multiple times and I tried to change the Clickhouse version for the latest but that don't solve this problem :/

This is not a clickhouse container, I don't use the docker compose file.
Maybe, there is something I miss while installing from source  ?

**Reply by vincentbernat** [reply]
Oh, OK, then you need to install the proto file to your ClickHouse installation. Check http://akvorado-orchestrator:8080/api/v0/orchestrator/clickhouse/init.sh for details about this.

**Reply by charlyforot** [reply]
You are right !

```
curl http://akvorado/api/v0/orchestrator/clickhouse/init.sh | sh
```

solved the problem.

Thank you !

**Reply by vincentbernat** [reply]
You may need to do that for each schema change (which may happens during upgrade too). For Protobuf, ClickHouse does not know how to retrieve the schema somewhere else than the file system.

**Comment by charlyforot** [comment, marked as answer]
```
curl http://akvorado/api/v0/orchestrator/clickhouse/init.sh | sh
```

thanks Vincent !
